### ISSUE TITLE
HDR / wide color hints

### ISSUE THREAD

**Issue by eeeps** [WICG/responsive-image-client-hints#24]
They might be nice! Especially if the files that take advantage of the extra display capabilities are significantly larger (>8 bits-per-channel) than 8-bit SDR/sRGB versions.

What could such hints contain? I see two options:

1. Lower-entropy versions of the hints would take their values from the relevant [media](https://www.w3.org/TR/mediaqueries-5/#dynamic-range) [queries](https://www.w3.org/TR/mediaqueries-5/#color-gamut).
2. Higher entropy versions of the hints would take their values from [the HDR Canvas Proposal](https://github.com/w3c/ColorWeb-CG/blob/master/hdr_html_canvas_element.md), although this would require three significant pieces to land, first:
	1. The HDR Canvas Proposal would need to be spec'd, achieve consensus, and implemented.
	2. A permissions gate for accessing detailed display information from this API would have to be implemented as [Permissions Policy](https://w3c.github.io/webappsec-permissions-policy/) HTTP headers (right now, the proposal only mentions user prompts).
	3. [Client Hints Infrastructure](https://wicg.github.io/client-hints-infrastructure/) would need to be updated, so that there were a new class of hints for which Permissions-Policies were checked/required even for first party requests.

All of that is making me lean towards the lower-entropy values, but more research on use cases is required.

**Comment by svgeesus** [comment]
> What are the use cases for those that the low entropy hints don't satisfy?

Knowing things like the max full-screen luminance allows accurate tone-mapping so that content on the Web looks as good as native content on the same platform. A simple SDR-or-HDR media query gives no such information, resulting in sub-standard generic mapping that fails to take advantage of the actual capabilities of the display (dull content, or harsh clipped highlights, or the 'battle of winterfell effect" where you can't see a thing)

**Comment by eeeps** [comment]
@svgeesus makes sense, though I worry about fingerprinting. (Also I'm curious to see some "generic mapping" failures; I'll try to hunt some down but feel free to share!).

Before we expose max full-screen luminance via Client Hints, [it should really be available somewhere else first](https://datatracker.ietf.org/doc/html/rfc8942#name-information-exposure) (via JS or CSS). Is https://github.com/w3c/ColorWeb-CG/blob/master/hdr_html_canvas_element.md still the thing to track for that?

**Comment by eeeps** [comment]
Answering my own question... based on [the recent Intent to Prototype](https://groups.google.com/a/chromium.org/g/blink-dev/c/S609KGTkSSk/m/1EW4CWn3BgAJ), it looks like that is the current proposal. And just making sure I understand the proposal correctly, the specific thing that would be useful to expose as a Client Hint would be <a href="https://github.com/w3c/ColorWeb-CG/blob/main/hdr_html_canvas_element.md#querying-screen-high-dynamic-range-parameters"><code>highDynamicRangeHeadroom</code></a>?

**Comment by svgeesus** [comment]
Yes, that is the actively maintained place to track

**Comment by svgeesus** [comment]
> he specific thing that would be useful to expose as a Client Hint would be <a href="https://github.com/w3c/ColorWeb-CG/blob/main/hdr_html_canvas_element.md#querying-screen-high-dynamic-range-parameters"><code>highDynamicRangeHeadroom</code></a>?

Exactly
